core/src/main/java/org/geysermc/geyser/entity/type/living/IronGolemEntity.java

```diff
@@ -45,6 +45,8 @@ public IronGolemEntity(GeyserSession session, int entityId, long geyserId, UUID
         setFlag(EntityFlag.BRIBED, true);
         // Required, or else the overlay is black
         dirtyMetadata.put(EntityData.COLOR_2, (byte) 0);
+        // Default max health. Ensures correct cracked texture is used
+        maxHealth = 100f;
```

Comment:
Just add to the comment here saying it can be reproduced on 1.19.0 Java Edition vanilla/Fabric by spawning a new instance of the mob.